app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -300,4 +336,44 @@ public void showNotificationGroup(long priority) {
                 (NotificationManager) this.getSystemService(Context.NOTIFICATION_SERVICE);
         notificationManager.notify(-5, b.build());
     }
+
+    private Bitmap getIcon(Integer appid) {
+
+        if(appid==-1){
+            return BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
+        }
+
+        Bitmap icon;
+        RequestCreator request;
+        try {
+            request= cache.load(Utils.resolveAbsoluteUrl(settings.url() + "/", appIdMap.get(appid)));
+            icon = request.get();
+        } catch (IOException e) {
+            icon = BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
+        }
+
+        return icon;
```

Comment:
Return only and only declare variables where they needed
```suggestion
        try {
            RequestCreator request= cache.load(Utils.resolveAbsoluteUrl(settings.url() + "/", appIdMap.get(appid)));
            return request.get();
        } catch (IOException e) {
            Log.e("Could not load image", e);
            return BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
        }
```